src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewComponents/DefaultViewComponentHelper.cs

```diff
@@ -130,19 +132,28 @@ private ViewComponentDescriptor SelectComponent(Type componentType)
                 componentType.FullName));
         }
 
-        private async Task<IHtmlContent> InvokeCoreAsync(
-            ViewComponentDescriptor descriptor,
-            object arguments)
+        // Internal for testing
+        internal IDictionary<string, object> GetArgumentDictionary(ViewComponentDescriptor descriptor, object arguments)
         {
+            if (descriptor.Parameters.Count == 1 && descriptor.Parameters[0].ParameterType.IsAssignableFrom(arguments.GetType()))
+            {
+                return new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase)
```

Comment:
Spoke offline. Not doing this because `PropertyHelper.ObjectToDictionary()` also uses `OrdinalIgnoreCase`.